src/lib/HST/Frontend/HSETransformation.hs

```diff
@@ -0,0 +1,485 @@
+module HST.Frontend.HSETransformation where
+
+import qualified Language.Haskell.Exts.Syntax  as HSE
+import qualified Language.Haskell.Exts.SrcLoc  as Src
+import qualified HST.Frontend.Syntax           as HST
+
+
+tfHSEtoHSTModule :: HSE.Module s -> HST.Module s (HSE.Literal s) (HSE.Type s)
+tfHSEtoHSTModule (HSE.Module _ _ _ _ decls) = HST.Module
+  (map tfHSEtoHSTDecl (filter supportedDecls decls))
+ where
+  supportedDecls (HSE.DataDecl _ _ _ _ _ _) = True
+  supportedDecls (HSE.FunBind _ _         ) = True
+  supportedDecls (HSE.PatBind _ _ _ _     ) = True
+  supportedDecls _                          = False
```

Comment:
One small problem with this would be that we wouldn't throw an error message in the `transformBinds` function when unsupported declarations occur within the declarations there, but we would simply throw these delarations away without notice.
We could add these error messages by using `map (fromMaybe (error "Unsupported declaration") . transformDecl) decls` instead of `mapMaybe` in `transformBinds`, but we wouldn't be able to show the user the unsupported declaration. I still think this would be the best solution, however.